app/app/models.py

```diff
@@ -92,6 +88,27 @@ def validate_studentNumber(self, key, studentNumber):
     @validates('unitCode')
     def validate_unitCode(self, key, unitCode):
         for i, c in enumerate(unitCode):
-            if (i < 4 and not c.isupper()) or (i >= 4 and not c.isnumeric()):
+            if (i < 4 and not (c.isupper() or c.islower())) or (i >= 4 and not c.isnumeric()):
                 raise ValueError("unitCode must be of the form CCCCNNNN")
         return unitCode
+
+    @validates('enquiry')
+    def validate_enquiry(self, key, enquiry):
+        if enquiry not in enquiryType:
+            raise ValueError('Enquiry is an invalid type')
+        else:
+            return enquiry
+
+    @validates('queue')
+    def validate_queue(self, key, queue):
+        if queue not in queueType:
+            raise ValueError('Enquiry is an invalid type')
+        else:
+            return queue
+
+    @validates('status')
+    def validate_status(self, key, status):
+        if status not in statusType:
+            raise ValueError('Enquiry is an invalid type')
+        else:
+            return status
```

Comment:
Looks good! May consider to add validate_studentName as well to make sure there is no numeric character in it.

Reply:
Good idea. Done!